apps/st2-packs/store.js

```diff
@@ -0,0 +1,244 @@
+          };
+        case 'error':
+          return {
+            ...state
+          };
+        default:
+          return {
+            ...state
+          };
+      }
+
+    case 'FETCH_PACK_INDEX':
+
+      switch(action.status) {
+        case 'success':
+          const packs = { ...state.packs };
+
+          _.forEach(action.payload, pack => {
+            packs[pack.ref] = {
+              ...state.packs[pack.ref],
+              ...pack
+            };
+          });
+
+          return {
+            ...state,
+            packs
+          };
+        case 'error':
+          return {
+            ...state
+          };
+        default:
+          return {
+            ...state
+          };
+      }
+
+      return {
+        ...state
+      };
+
+      case 'FETCH_PACK_CONFIG_SCHEMAS':
+
+        switch(action.status) {
+          case 'success':
+            const packs = { ...state.packs };
+
+            _.forEach(action.payload, pack => {
+              packs[pack.ref] = {
+                ...state.packs[pack.ref],
+                ...pack
+              };
+            });
+
+            return {
+              ...state,
+              packs
+            };
+          case 'error':
+            return {
+              ...state
+            };
+          default:
+            return {
+              ...state
+            };
+        }
+
+        return {
+          ...state
+        };
+
+      case 'FETCH_PACK_CONFIGS':
+
+        switch(action.status) {
+          case 'success':
+            const packs = { ...state.packs };
+
+            _.forEach(action.payload, pack => {
+              packs[pack.ref] = {
+                ...state.packs[pack.ref],
+                ...pack
+              };
+            });
+
+            return {
+              ...state,
+              packs
+            };
+          case 'error':
+            return {
+              ...state
+            };
+          default:
+            return {
+              ...state
+            };
+        }
+
+        return {
+          ...state
+        };
+
+    case 'SELECT_PACK':
+      const { ref } = action;
+
+      return {
+        ...state,
+        selected: ref || Object.keys(state.packs)[0]
+      };
+
+    case 'SET_FILTER':
+      const { filter } = action;
+
+      return {
+        ...state,
+        filter
+      };
+
+    default:
+      return state;
+  }
+};
+
+const promiseMiddleware = () => next => action => {
+  if (!action.promise) {
+    return next(action);
+  }
+
+  function actionFactory(status, data) {
+    const { promise, ...newAction } = action;
+    return {
+      ...newAction,
+      status,
+      ...data
+    };
+  }
+
+  next(actionFactory());
+  return action.promise.then(
+    payload => next(actionFactory('success', { payload })),
+    error => next(actionFactory('error', { error }))
+  );
+};
+
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const store = createStore(
```

Comment:
There is a need in that sort of moves, but I'd rather keep them out of this particular PR. As soon as we start an active phase of transitioning from Angular to React, I see us moving lots of components (including store) to higher levels, but I'm afraid it might create additional clutter so I'd prefer we do it in one sweep.